libdrgn/debug_info.c

```diff
@@ -4324,7 +4325,8 @@ struct drgn_error *open_elf_file(const char *path, int *fd_ret, Elf **elf_ret)
 	*fd_ret = open(path, O_RDONLY);
 	if (*fd_ret == -1)
 		return drgn_error_create_os("open", errno, path);
-	*elf_ret = dwelf_elf_begin(*fd_ret);
+	// *elf_ret = dwelf_elf_begin(*fd_ret);
+	*elf_ret = elf_begin(*fd_ret, ELF_C_READ_MMAP_PRIVATE, NULL);
```

Comment:
Looks like `dwelf_elf_begin()` was added in 0.175. We can probably define a stub like:

```c
#if !_ELFUTILS_PREREQ(0, 175)
static Elf *dwelf_elf_begin(int fd)
{
	return elf_begin(fd, ELF_C_READ_MMAP_PRIVATE, NULL);
}
#endif
```

(With a comment about what the real `dwelf_elf_begin()` does that `elf_begin()` doesn't.)

This might need an `#include <elfutils/version.h>`. It can probably go in `libdrgn/debug_info.c`. A little ugly but not the worst.